### PR TITLE
Using a separate Package class for resolved packages

### PR DESCRIPTION
**Pull request by akshita31**
Renamed the existing 'Package' interface to `PackageJSONPackage` and created a new `Package` class which is formed by resolving the paths using the formed extension paths, which is then used for the download and installl process

### CODE REVIEW

File: src/CSharpExtDownloader.ts [hunk 1]
```diff
@@ -54,9 +54,10 @@ export class CSharpExtDownloader {
     }
 }
 
-export function GetRunTimeDependenciesPackages(packageJSON: any): Package[] {
+export function GetRunTimeDependenciesPackages(packageJSON: any, extensionPath: string): PackageJSONPackage[] {
     if (packageJSON.runtimeDependencies) {
-        return JSON.parse(JSON.stringify(<Package[]>packageJSON.runtimeDependencies));
+        let dependencies = <PackageJSONPackage[]>packageJSON.runtimeDependencies;
+        return dependencies.map(pkg => new Package(pkg, extensionPath));
```
Comment (TheRealPiotrP) [inline src/CSharpExtDownloader.ts]: I don't get it. This method returns an array of `Package`s. So why is the return type an array of `PackageJSONPackage`?

File: src/omnisharp/OmnisharpDownloader.ts [hunk 2]
```diff
@@ -42,6 +43,12 @@ export class OmnisharpDownloader {
         }
     }
 
+    private GetResolvedPackagesFromVersion(version: string, packageJSON: any, serverUrl: string, installPath: string, extensionPath: string) {
```
Comment (TheRealPiotrP) [inline src/omnisharp/OmnisharpDownloader.ts]: I think this should be `GetResolvedPackagesForVersion`

File: src/CSharpExtDownloader.ts (hunk 1 above)
Comment (akshita31) [reply src/CSharpExtDownloader.ts]: Yes you are right. Will correct that.

File: src/omnisharp/OmnisharpPackageCreator.ts [hunk 3]
```diff
@@ -3,24 +3,28 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { Package } from "../packageManager/Package";
+import { PackageJSONPackage } from "../packageManager/PackageJSONPackage";
```
Comment (TheRealPiotrP) [inline src/omnisharp/OmnisharpPackageCreator.ts]: There should be exactly one component that knows/cares what a PackageJSONPackage is. Why do we need to look at the Package.json in two places?

**Comment by akshita31** [comment]
I think we need to redo some things here for a better shape. Closing this for now.

**Pull request closed without merging**